python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -330,24 +351,132 @@ cdef class Column:
             c_result = make_column_from_scalar(dereference(slr.get()), size)
         return Column.from_libcudf(move(c_result))
 
+    @staticmethod
+    cdef Column from_gpumemoryview(
+        gpumemoryview data,
+        tuple shape,
+        tuple strides,
+        DataType dtype,
+    ):
+        """
+        Construct a Column from a gpumemoryview and array metadata.
+
+        Parameters
+        ----------
+        data : gpumemoryview
+            The data buffer.
+        shape : tuple
+            The shape of the array.
+        strides : tuple or None
+            The strides of the array, or None if C-contiguous.
+        dtype : DataType
+            The element type of the array.
+
+        Returns
+        -------
+        Column
+        """
+        if not is_c_contiguous(shape, strides, size_of(dtype)):
+            raise ValueError("Data must be C-contiguous")
+
+        if len(shape) == 1:
+            size = shape[0]
+            return Column(dtype, size, data, None, 0, 0, [])
+
+        elif len(shape) == 2:
+            num_rows, num_cols = shape
+
+            if num_rows >= numeric_limits[size_type].max():
+                raise ValueError(
+                    "Number of rows exceeds size_type limit for offsets column."
+                )
+
+            offsets_col = sequence(
+                num_rows + 1,
+                Scalar.from_py(0, DataType(type_id.INT32)),
+                Scalar.from_py(num_cols, DataType(type_id.INT32)),
+            )
+
+            flat_size = num_rows * num_cols
+
+            data_col = Column(
+                data_type=dtype,
+                size=flat_size,
+                data=data,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[],
+            )
+
+            return Column(
+                data_type=DataType(type_id.LIST),
+                size=num_rows,
+                data=None,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[offsets_col, data_col],
+            )
+
+        else:
+            raise ValueError("Only 1D or 2D arrays are supported")
```

Comment:
Just noting crossfit also has a requirement of transforming a 3D cupy array into a `ListColumn`: https://github.com/rapidsai/crossfit/blob/c8ba17a72dc2768651e372f08f77afdd79f24070/crossfit/backend/cudf/series.py#L110

So eventually it would be nice to generalize collapsing any ND array into a `ListColumn`

Reply:
Doing this in a follow up